modules/mod_manager.ts

```diff
@@ -330,22 +397,77 @@ class ModManager {
         await config.SaveConfig(Main.config);
+        
+        let setup_func: string = "";
+        if((typeof(collectionVersion) != "undefined") && (!(IsInstall(this.source_manager.data)))) {
+            if (!(typeof (this.source_manager.data[FindCollectionNumber(this.source_manager.data, collectionVersion)].setupfunc) == "undefined")) {
+                setup_func = this.source_manager.data[FindCollectionNumber(this.source_manager.data, collectionVersion)].setupfunc;
+            }
+        }
+        else {
+            if(IsInstall(this.source_manager.data)) {
+                if (!(typeof (this.source_manager.data.setupfunc) == "undefined")) {
+                    setup_func = this.source_manager.data.setupfunc;
+                }
+            }
+        }
+        switch (setup_func) {
+            case "movecfgs":
+                ElectronLog.log("Executing movecfgs install operation")
+                let filesToMove = ["autoexec.cfg", "scout.cfg", "soldier.cfg", "pyro.cfg", "demoman.cfg", "heavyweapons.cfg",
+                "engineer.cfg", "medic.cfg", "sniper.cfg", "spy.cfg"]
+                //Move cfgs
+                let cfgpath = path.join(Main.config.tf2_directory, "tf", "cfg");
+                let usercfgpath = path.join(cfgpath, "user");
+                ElectronLog.log("cfg path is: " + cfgpath);
+                ElectronLog.log("user cfg path is: \"" + usercfgpath + "\"");
+                if (!(fs.existsSync(usercfgpath))) {
+                    //Create usercfgpath
+                    ElectronLog.log("user cfg path does not exist, creating...");
+                    fs.mkdirSync(usercfgpath, {recursive: true});
+                }
+                //Actually move them
+                fs.readdir(cfgpath, (err, files) => {
+                    if (err) {
+                        ElectronLog.error("there was an error on mod_manager.ts, line 447 when trying to read the cfg directory to move cfg files. Trying to read directory: " + cfgpath + " , the error was:" + err)
+                    }
+                    else {
+                        files.forEach(file => {
+                            ElectronLog.log("Checking if should move file: " + file)
+                            if(filesToMove.indexOf(file) > -1) {
+                                ElectronLog.log("Trying to move file " + path.join(cfgpath, file) + " to " + path.join(usercfgpath, path.basename(file)))
+                                fs.rename(path.join(cfgpath ,file), path.join(usercfgpath, path.basename(file)), (err) => {
+                                    if (err) {
+                                        ElectronLog.error("There was an error trying to move file " + path.join(cfgpath, file) + " to " + path.join(usercfgpath, path.basename(file)))
+                                    }
+                                })
+                            }
+                        })
+                    }
+                })
+                break;
```

Comment:
I like this system to allow configurable post install tasks.
I think a better way is to add a base method to the mod sources, and let that handle these tasks instead of them being done here.
Again i think the many type checks/casts are not good.

How about `ModInstallSource` has a method to handle a PostInstallTask, and this logic can be moved over, and all mod configurations can support this functionality.